src/python/pants/backend/docker/goals/package_image_integration_test.py

```diff
@@ -61,3 +63,37 @@ def test_docker_build(rule_runner) -> None:
     assert "Built docker image: test-image:1.0" == result.artifacts[0].extra_log_lines[0]
     assert "Docker image ID:" in result.artifacts[0].extra_log_lines[1]
     assert "<unknown>" not in result.artifacts[0].extra_log_lines[1]
+
+
+def test_docker_build_with_args_and_labels(rule_runner) -> None:
+    """This test requires a running docker daemon."""
+    rule_runner.write_files(
+        {
+            "src/BUILD": dedent(
+                """\
+                docker_image(
+                    name='test-ubuntu',
+                    extra_build_args=['UBUNTU_VERSION=20.04'],
+                    image_labels={
+                        'ubuntu': '{build_args.UBUNTU_VERSION}',
+                        'base_image': '{tags.baseimage}'
```

Comment:
If I comment this out, the test passes. So even though the error messages says the `KeyError` on `UBUNTU_VERSION` is the cause, the real cause seems to be:
```
pants.backend.docker.goals.package_image.DockerImageOptionValueError: Invalid value for the `image_labels` field of the `docker_image` target at src:test-ubuntu: '{tags.baseimage}'.
```